### DISCUSSION TITLE
Is keyword_length = 0 no longer supported?

### DISCUSSION THREAD

**Discussion by sQVe** [hrsh7th/nvim-cmp#1358] [Q&A] [answered]
👋🏼,

I've been using `keyword_length = 0` when manually triggering completion at times but it is no longer working. Seeing some issues and discussion mentioning the same.

Any information on this?

**Comment by Shougo** [comment]
I think the feature is not supported.

**Reply by sQVe** [reply]
That's a shame, since being able to list everything is really useful in certain situations. 

I typically use it to list the API when importing a package. Now I would either have to go to their documentation or just wing it.

**Comment by fitrh** [comment, marked as answer]
What is the purpose of `keyword_length = 0`? I mean, we can list everything by calling `complete()`

**Reply by sQVe** [reply]
@fitrh I'm not able to list everything by calling `complete()`, so that's the reason for using `keyword_length = 0`.

**Reply by sQVe** [reply]
Are you really able to list everything when calling `complete()`?

**Reply by fitrh** [reply]
> Are you really able to list everything when calling `complete()`?

I think so, what LSP are you using?

This is with svelteserver + tsserver, note that I've disabled the scrollbar and set the `pumheight` to 9

![20221221_174525](https://user-images.githubusercontent.com/55179750/208874665-146ed60f-a2e5-4db6-9a57-7d54de3c2c4f.jpeg)

This is when `pumheight = 0`

![20221221_175407](https://user-images.githubusercontent.com/55179750/208876783-de3d6183-fbf2-4871-a6b8-6bf958e944fa.jpeg)

![20221221_175812](https://user-images.githubusercontent.com/55179750/208877182-8c7a5f57-fc04-4aed-93ad-481b7870a8e4.jpeg)

**Reply by sQVe** [reply]
I'm mostly using `tsserver`. Weird that it works for you 🤔

**Reply by sQVe** [reply]
Found the issue. If I remove the following https://github.com/sQVe/dotfiles/blob/cd741918c63404f7f07a011f5d2d13c7a6344d9b/config/nvim/lua/sQVe/plugins/cmp.lua#L131 it shows all when triggering `complete()` 🤔

**Reply by fitrh** [reply]
> Found the issue. If I remove the following https://github.com/sQVe/dotfiles/blob/cd741918c63404f7f07a011f5d2d13c7a6344d9b/config/nvim/lua/sQVe/plugins/cmp.lua#L131 it shows all when triggering `complete()` thinking

I think it should be `cmp.ContextReason.Manual`
